lib/matplotlib/collections.py

```diff
@@ -1966,6 +1966,13 @@ def set_paths(self):
     def get_datalim(self, transData):
         return (self.get_transform() - transData).transform_bbox(self._bbox)
 
+    def set_array(self, A):
+        # Allow QuadMesh.set_array(A) to accept 2d input
+        # as long is it is the same size as the current 1d data
+        if A.ndim == 2 and A.size == self._A.size:
```

Comment:
It might make sense to actually issue a warning here if `A.size != self._A.size` to address some common issues of the edges (n+1, m+1) compared to the data (n, m). Thoughts?

Reply:
I'm a little confused as to why A needs to be raveled at all here?  Why does QuadMesh store its array as 1-D?